trestlebot/entrypoints/init.py

```diff
@@ -0,0 +1,216 @@
+from trestlebot.entrypoints.log import set_log_level_from_args
+from trestlebot.tasks.authored import types as model_types
+
+
+logger = logging.getLogger(__name__)
+logging.getLogger("trestle.core.commands.init").setLevel("CRITICAL")
+
+OSCAL_MODEL_SSP = model_types.AuthoredType.SSP.value
+OSCAL_MODEL_COMPDEF = model_types.AuthoredType.COMPDEF.value
+
+
+class InitEntrypoint:
+    """Entrypoint for the init command."""
+
+    TEMPLATES_MODULE = "trestlebot.entrypoints.templates"
+    SUPPORTED_PROVIDERS = [GITHUB, GITLAB]
+    SUPPORTED_MODELS = [
+        OSCAL_MODEL_SSP,
+        OSCAL_MODEL_COMPDEF,
+    ]
+    PROVIDER_TEMPLATES = {
+        GITHUB: {
+            OSCAL_MODEL_SSP: [
+                "trestlebot-autosync-catalog.yml",
+                "trestlebot-autosync-profile.yml",
+            ],
+            OSCAL_MODEL_COMPDEF: [
+                "trestlebot-create-component-definition.yml",
+                "trestlebot-autosync-catalog.yml",
+                "trestlebot-autosync-profile.yml",
+                "trestlebot-rules-transform.yml",
+            ],
+        }
+    }
+    MODEL_DIRS = {
+        OSCAL_MODEL_SSP: [
+            "system-security-plans",
+            "component-definitions",
+            "catalogs",
+            "profiles",
+        ],
+        OSCAL_MODEL_COMPDEF: [
+            "component-definitions",
+            "catalogs",
+            "profiles",
+            "rules",
+        ],
+    }
+
+    def __init__(self, parser: argparse.ArgumentParser) -> None:
+        self.parser: argparse.ArgumentParser = parser
+        self.setup_init_arguments()
+
+    def setup_init_arguments(self) -> None:
+        """Setup arguments for the init entrypoint."""
+        self.parser.add_argument(
+            "-v",
+            "--verbose",
+            help="Display verbose output",
+            action="count",
+            default=0,
+        )
+        self.parser.add_argument(
+            "--working-dir",
+            type=str,
+            required=False,
+            default=".",
+            help="Working directory wit git repository",
+        )
+        self.parser.add_argument(
+            "--provider",
+            required=False,
+            type=str,
+            choices=self.SUPPORTED_PROVIDERS,
+            default="github",
+            help="Name of CI/CD provider",
+        )
+        self.parser.add_argument(
+            "--oscal-model",
+            required=True,
+            type=str,
+            choices=self.SUPPORTED_MODELS,
+            help="OSCAL model type to run tasks on.",
+        )
+
+    def _call_trestle_init(self, args: argparse.Namespace) -> None:
+        """Call compliance-trestle to initialize workspace"""
+        trestle_args = argparse.Namespace(
+            verbose=args.verbose,
+            trestle_root=pathlib.Path(args.working_dir),
+            full=False,
+            govdocs=True,
+            local=False,
+        )
+        InitCmd()._run(trestle_args)
```

Comment:
Might be worth checking the error code here. `trestle` store that information [here](https://github.com/oscal-compass/compliance-trestle/blob/6c2d3f3bd8d6eeaf04e0a931ce39b8b52646e95a/trestle/core/commands/common/return_codes.py#L36).